pkg/substitution/substitution_test.go

```diff
@@ -174,6 +174,22 @@ func TestValidateVariablePs(t *testing.T) {
 			vars:   sets.NewString("baz", "foo"),
 		},
 		expectedError: nil,
+	}, {
+		name: "valid usage of an individual attribute of an object param",
+		args: args{
+			input:  "--flag=$(params.objectParam.key1)",
```

Comment:
Maybe worths adding more valid tests like `params.['objectParam'].key1` and `params.["objectParam"].key1`

Reply:
> Maybe worths adding more valid tests like `params.['objectParam'].key1` and `params.["objectParam"].key1`

I think both `params.['objectParam'].key1` and `params.["objectParam"].key1` are invalid reference for object keys according to the [tep](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#variable-replacement-with-object-params).

Reply:
> I think both params.['objectParam'].key1 and params.["objectParam"].key1 are invalid reference for object keys according to the [tep](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#variable-replacement-with-object-params).

Ah why is that? I the [tep](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#variable-replacement-with-object-params) only covers several examples for referring to keys in params. Ideally we should extend the key references based on what we have for [params](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-pipeline)

Reply:
For existing bracket notation, the [doc](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-pipeline) seems only mentioning something like `params["<NAME>"]` instead of `params.["<NAME>"]`. [TEP-0080](https://github.com/tektoncd/community/blob/main/teps/0080-support-domainscoped-parameterresult-names.md#proposal) added support for bracket notation, which is meant to support names that contain dots (.).


If we use bracket notation for object, it seems very confusing to users i.e. 
- `params['objectParam'][*]` (array type seems not supporting this?)
- `params['objectParam'].key1` (also do we want to consider `params['objectParam']['key1']`).  

So I'd lean more towards what the [tep-75](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#variable-replacement-with-object-params) suggested (at least initially).

Reply:
> For existing bracket notation, the [doc](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-pipeline) seems only mentioning something like params["<NAME>"] instead of params.["<NAME>"]. [TEP-0080](https://github.com/tektoncd/community/blob/main/teps/0080-support-domainscoped-parameterresult-names.md#proposal) added support for bracket notation, which is meant to support names that contain dots (.).

Ah my bad that there is a typo in the comment. I meant to say `params['objectParam'].key1` and `params["objectParam"].key1`

Reply:
> > For existing bracket notation, the [doc](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-pipeline) seems only mentioning something like params[""] instead of params.[""]. [TEP-0080](https://github.com/tektoncd/community/blob/main/teps/0080-support-domainscoped-parameterresult-names.md#proposal) added support for bracket notation, which is meant to support names that contain dots (.).

Ah my bad that there is a typo in the comment. I meant to say `params['objectParam'].key1` and `params["objectParam"].key1`

> params['objectParam']['key1']

Yes I think this should be added

Reply:
This is out of the scope of this pr. Let's discuss offline later about whether we want to support chained bracket notation for object name (and keys).